loadable_apps/loadable_sample/wifiapp/binary_update.c

```diff
@@ -166,7 +166,7 @@ static int binary_update_download_binary(binary_update_info_t *binary_info, int
 		fail_cnt++;
 	}
 
-	return ERROR;
+	return ret;
```

Comment:
This causes mis-operation for failure case. When write_fd is failed to open, ret has a positive value. Then that will not increase the fail_cnt. Please check.